kcidb/test_db.py

```diff
@@ -19,194 +17,201 @@ def test_schemas_main():
+        }
+    )
+    data = empty_database.dump()
 
-    driver_source = textwrap.dedent(f"""
-        from unittest.mock import patch, Mock
-        client = Mock()
-        client.dump_iter = Mock(return_value=iter({repr((empty, empty))}))
-        with patch("kcidb.db.Client", return_value=client) as \
-                Client:
-            status = function()
-        Client.assert_called_once_with("bigquery:project.dataset")
-        client.dump_iter.assert_called_once()
-        return status
-    """)
-    assert_executes("", *argv, driver_source=driver_source,
-                    stdout_re=re.escape(json.dumps(empty) + "\n" +
-                                        json.dumps(empty) + "\n"))
+    assert_executes("", *argv,
+                    stdout_re=re.escape(json.dumps(data) + "\n"))
 
 
-def test_query_main():
+def test_query_main(empty_database):
     """Check kcidb-db-query works"""
-    driver_source = textwrap.dedent("""
-        from unittest.mock import patch
-        with patch("kcidb.db.Client"):
-            return function()
-    """)
-    argv = ["kcidb.db.query_main", "-d", "bigquery:project.dataset"]
-    assert_executes("", *argv, driver_source=driver_source)
+
+    empty_database.is_initialized()
+    argv = ["kcidb.db.query_main", "-d", empty_database.database]
+    assert_executes("", *argv)
 
     argv = [
-        "kcidb.db.query_main", "-d", "bigquery:project.dataset",
+        "kcidb.db.query_main", "-d", empty_database.database,
         "-c", "test:checkout:1", "-b", "test:build:1",
         "-t", "test:test:1",
         "--parents", "--children", "--objects-per-report", "10",
         "--indent=0",
     ]
+
+    empty_database.load(dict(
+        version=dict(major=4, minor=1),
+        checkouts=[
+            dict(id="_:1", origin="_"),
+            dict(id="_:2", origin="_"),
+        ],
+        builds=[
+            dict(id="_:1", origin="_", checkout_id="_:1"),
+            dict(id="_:2", origin="_", checkout_id="_:2"),
+        ],
+        tests=[
+            dict(id="_:1", origin="_", build_id="_:1"),
+            dict(id="_:2", origin="_", build_id="_:2"),
+        ],
+        issues=[
+            dict(id="_:1", origin="_", version=1),
+            dict(id="_:2", origin="_", version=1),
+            dict(id="_:3", origin="_", version=1),
+            dict(id="_:4", origin="_", version=1),
+        ],
+        incidents=[
+            dict(id="_:1", origin="_", issue_id="_:1", issue_version=1,
+                 build_id="_:1"),
+            dict(id="_:2", origin="_", issue_id="_:2", issue_version=1,
+                 test_id="_:1"),
+            dict(id="_:3", origin="_", issue_id="_:3", issue_version=1,
+                 build_id="_:2", test_id="_:2"),
+            dict(id="_:4", origin="_", issue_id="_:4", issue_version=1,
+                 test_id="_:1"),
+            dict(id="_:5", origin="_", issue_id="_:4", issue_version=1,
+                 test_id="_:2"),
+        ],
+    ))
```

Comment:
Please also reduce the data to just one object per type. Then you wouldn't have a problem comparing it to the query result due to order changing. You can also just create a single fixture with the data, shared by several tests and use it here.